conans/client/toolchain/make.py

```diff
@@ -0,0 +1,199 @@
+                    else ifeq ($(CONAN_TC_LIBCXX),libc++)
+                        CONAN_TC_CXXFLAGS += -stdlib=libc++
+                    endif
+                else ifeq ($(CONAN_TC_COMPILER),sun-cc)
+                    ifeq ($(CONAN_TC_LIBCXX),libCstd)
+                        CONAN_TC_CXXFLAGS += -library=Cstd++
+                    else ifeq ($(CONAN_TC_LIBCXX),libstdcxx)
+                        CONAN_TC_CXXFLAGS += -library=stdcxx4
+                    else ifeq ($(CONAN_TC_LIBCXX),libstlport)
+                        CONAN_TC_CXXFLAGS += -library=stlport4
+                    else ifeq ($(CONAN_TC_LIBCXX),libstdc++)
+                        CONAN_TC_CXXFLAGS += -library=stdcpp
+                    endif
+                endif
+                ifeq ($(CONAN_TC_LIBCXX),libstdc++11)
+                    CONAN_TC_DEFINES += GLIBCXX_USE_CXX11_ABI=1
+                else ifeq ($(CONAN_TC_LIBCXX),libstdc++)
+                    CONAN_TC_DEFINES += GLIBCXX_USE_CXX11_ABI=0
+                endif
+            endif
+            ifeq ($(CONAN_TC_BUILD_TYPE),Release)
+                CONAN_TC_DEFINES += NDEBUG
+            endif
+            ifeq ($(CONAN_TC_SET_FPIC),True)
+                CONAN_TC_CFLAGS += -fPIC
+                CONAN_TC_CXXFLAGS += -fPIC
+                CONAN_TC_SHARED_LINKER_FLAGS += -fPIC
+            endif
+        endif
+
+        CONAN_TC_CPPFLAGS += $(addprefix -D,$(CONAN_TC_DEFINES))
+
+        # Call this function in your Makefile to have Conan variables added to the standard variables
+        # Example:  $(call CONAN_TC_SETUP)
+
+        CONAN_TOOLCHAIN_SETUP =  \
+            $(eval CFLAGS += $(CONAN_TC_CFLAGS)) ; \
+            $(eval CXXFLAGS += $(CONAN_TC_CXXFLAGS)) ; \
+            $(eval CPPFLAGS += $(CONAN_TC_CPPFLAGS)) ; \
+        )
+
+    """)
+
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+
+        self._set_fpic = self._deduce_fpic()
+        self._set_libcxx = True
+
+        self._compiler = conanfile.settings.get_safe("compiler")
+        self._compiler_version = conanfile.settings.get_safe("compiler.version")
+        self._compiler_runtime = conanfile.settings.get_safe("compiler.runtime")
+        self._libcxx = conanfile.settings.get_safe("compiler.libcxx")
+
+        # cpp standard
+        self._cppstd = cppstd_from_settings(conanfile.settings)
+        self.cppstd_flag = cppstd_flag(conanfile.settings)
+
+        self._os_host = conanfile.settings.get_safe("os")
+        self._arch_host = conanfile.settings.get_safe("arch")
+        self._os_target, self._arch_target = get_target_os_arch(conanfile)
+        self._arch_build, self._os_build = self._get_build_os_arch()
+        self._build_type = conanfile.settings.get_safe("build_type")
+        # Precalculate build, host, target triplets
+        self._trip_build, self._trip_host, self._trip_target = self._get_host_build_target_flags()
+
+        self.definitions = {}
+
+    def _deduce_fpic(self):
```

Comment:
Duplicated in `toolchain/cmake.py` (hopefully exactly the same lines, but...), maybe it is time to generalize this logic and move it to a proper location as discussed here (https://github.com/conan-io/conan/pull/7076/files#diff-1996aeebc7b0eebed4dd936b368d712b) cc/ @memsharded

Reply:
Ok, i will make an issue to refactor in next release.

Reply:
Looks like there's a better version of this body of logic here: 
https://github.com/conan-io/conan/blob/develop/conans/client/conanfile/configure.py#L48